### PR TITLE
[Closes #46] Added ktn_list:map/3 and tests (100%) coverage of the module

### LINKED ISSUE #46: Add a ktn_lists:map/3 function
This function would allow to specify a list of additional arguments to the `Fun` supplied thus avoiding the need to create an anonymous `fun` that captures closures, which is pretty common when calling `lists:map`.

```
- spec 'map'(fun(), list(), list()) -> list(). 
'map'(Fun, Args, List) ->
 %% TODO: Implementation
 ok.
```

`Fun` will be applied in the following way `apply(Fun, [Item | Args])`.

### PR DESCRIPTION
**Pull request by jfacorro**
#46

### CODE REVIEW

**Comment by gadgetci** [comment]
According to **Dialyzer**:
> 
dialyzer: Analysis failed with error:
No .beam files to analyze (no --src specified?)

File: src/ktn_lists.erl [hunk 1]
```diff
@@ -46,3 +47,13 @@ split_when(When, [Head | Tail], [Current0 | Rest]) ->
                      [Current | Rest]
              end,
     split_when(When, Tail, Result).
+
+%% @doc Like lists:map/2 but allows specifying additional arguments.
+%%      E.g.
+%%        ktn_listsmap(fun (X, Y) -> X + Y end, [2], [1, 2, 3]) = [2, 4, 6]
```
Comment (amilkr) [inline src/ktn_lists.erl]: You're missing the ':' between ktn_lists and map

Comment (amilkr) [inline src/ktn_lists.erl]: I would add a happier example where you use the additional arguments.

Comment (amilkr) [inline src/ktn_lists.erl]: The result is [3, 4, 5], isn't?

**Pull request merged**